packages/realm-server/main.ts

```diff
@@ -175,33 +175,33 @@ if (distURL) {
 
     let realmPermissions = getRealmPermissions(url);
 
-    realms.push(
-      new Realm(
-        {
-          url,
-          adapter: new NodeAdapter(resolve(String(path))),
-          loader,
-          indexRunner: getRunner,
-          runnerOptsMgr: manager,
-          getIndexHTML: async () =>
-            readFileSync(join(distPath, 'index.html')).toString(),
-          matrix: { url: new URL(matrixURL), username, password },
-          realmSecretSeed: REALM_SECRET_SEED,
-          permissions: realmPermissions.users,
-        },
-        {
-          deferStartUp: true,
-          ...(useTestingDomain
-            ? {
-                useTestingDomain,
-              }
-            : {}),
-        },
-      ),
+    let realm = new Realm(
+      {
+        url,
+        adapter: new NodeAdapter(resolve(String(path))),
+        loader,
+        indexRunner: getRunner,
+        runnerOptsMgr: manager,
+        getIndexHTML: async () =>
+          readFileSync(join(distPath, 'index.html')).toString(),
+        matrix: { url: new URL(matrixURL), username, password },
+        realmSecretSeed: REALM_SECRET_SEED,
+        permissions: realmPermissions.users,
+      },
+      {
+        deferStartUp: true,
+        ...(useTestingDomain
+          ? {
+              useTestingDomain,
+            }
+          : {}),
+      },
     );
+    realms.push(realm);
+    virtualNetwork.mount(realm.maybeExternalHandle);
   }
 
-  let server = new RealmServer(realms, {
+  let server = new RealmServer(realms, virtualNetwork, {
```

Comment:
I may be reading it wrong, but it seems like the realm server is already mounting it's maybeExternalHandle method, so why do we also mount it on line 201?

Reply:
Hmm, I see that on line 201 it's the only time the realm server is mounting `maybeExternalHandle`. The realm server is mounting `maybeHandle` - did you mean that?